zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -432,46 +475,103 @@ private void saveToFile() throws IOException {
     return new ArrayList<>(Interpreter.registeredInterpreters.values());
   }
 
+
+  public boolean findDefaultInterpreter(List<InterpreterInfo> infos) {
+    for (InterpreterInfo interpreterInfo : infos) {
+      if (interpreterInfo.isDefaultInterpreter())
+        return true;
+    }
+    return false;
+  }
+
+  public InterpreterSetting createNewSetting(String name, String refName,
+      List<Dependency> dependencies, InterpreterOption option, Properties p) {
+    InterpreterSetting setting = createFromInterpreterSettingRef(refName);
+    setting.setName(name);
+    setting.setRefName(refName);
+    setting.appendDependencies(dependencies);
+    setting.setInterpreterOption(option);
+    setting.updateProperties(p);
+    setting.setInterpreterGroupFactory(this);
+    interpreterSettings.put(setting.getId(), setting);
+    return setting;
+  }
+
+  private InterpreterSetting add(String refName,
+      InterpreterInfo interpreterInfo, Properties properties, String path)
+      throws InterpreterException, IOException, RepositoryException {
+    ArrayList<InterpreterInfo> infos = new ArrayList<>();
+    infos.add(interpreterInfo);
+    return add(refName, infos, new ArrayList<Dependency>(), defaultOption, properties, path);
+  }
+
   /**
-   * @param name       user defined name
-   * @param groupName  interpreter group name to instantiate
+   * @param refName    user defined name
```

Comment:
I think variable name and comment is bit confusing here.
`InterpreterSetting.name` holds user defined name and `refName` holds interpreter group name.
(that's why i think we can just keep `group` instead of `refName`)